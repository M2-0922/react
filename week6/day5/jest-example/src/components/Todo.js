import React from 'react'

const Todo = ({ todo }) => {
  const { id, title, completed } = todo;
  const h1 = <h1>todo - {title}</h1>;
  const text = completed ? <strike>{h1}</strike> : h1;
  return (
    <div data-testid={`todo-${id}`}>{text}</div>
  )
}

export default Todo