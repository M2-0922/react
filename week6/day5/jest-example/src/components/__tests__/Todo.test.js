import { render, screen, cleanup } from "@testing-library/react";
import Todo from "../Todo";
// snapshot
import renderer from "react-test-renderer";

// test("should render todo component", () => {
//     render(<Todo />)
//     const todoElement = screen.getByTestId("todo-1");
//     expect(todoElement).toBeInTheDocument();
// });

test("should render non-completed todo element", () => {
    const todo = { id: 2, title: "create home component", completed: false };
    render(<Todo todo={todo} />);
    const todoElement = screen.getByTestId("todo-2");
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent("create home component");
    expect(todoElement).not.toContainHTML("<strike><h1>create home component</h1></strike>")
});

test("should render completed todo element", () => {
    const todo = { id: 1, title: "create react app", completed: true };
    render(<Todo todo={todo} />);
    const todoElement = screen.getByTestId("todo-1");
    expect(todoElement).toContainHTML("<strike><h1>todo - create react app</h1></strike>")
});

test("matches snapshot", () => {
    const todo = { id: 3, title: "drink coffee", completed: true };
    const tree = renderer.create(<Todo todo={todo} />).toJSON();
    expect(tree).toMatchSnapshot();
})