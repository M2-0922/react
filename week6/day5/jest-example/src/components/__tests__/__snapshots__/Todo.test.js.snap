// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="todo-3"
>
  <strike>
    <h1>
      todo - 
      drink coffee
    </h1>
  </strike>
</div>
`;
